src/content.js

```diff
@@ -118,45 +127,55 @@ async function getPageMetadata(contentDir) {
   return pages;
 }
 
-async function buildMetaIndex(contentDir, invaderDefsDir, baseUrl, packageVersion) {
-  const pages = await getPageMetadata(contentDir);
-  const invaderStructDefs = await getInvaderStructDefs(invaderDefsDir);
-  const data = buildData(invaderStructDefs);
-
-  const findTagPageByName = (tagName) => {
-    const page = pages.find(page => page.tagName == tagName);
-    if (!page) throw new Error(`Failed to find tag page for ${tagName}`);
-    return page;
-  };
-
-  const resolvePage = (slug, headingId) => {
-    const page = pages.find(page => page._slug == slug && (!headingId || page._headers.find(hdg => hdg.id == headingId)));
-    if (!page) throw new Error(`Failed to find page with slug '${slug}' and headingId '${headingId}'`);
-    return page;
-  }
+async function renderContent(pages, data, buildOpts) {
+  const searchDocs = await Promise.all(pages.map(async (page) => {
+
+    const findTagPageByName = (tagName) => {
+      const otherPage = pages.find(otherPage => otherPage.tagName == tagName);
+      if (!otherPage) throw new Error(`Failed to find tag page for ${tagName}`);
+      return otherPage;
+    };
+
+    const resolvePage = (pathTail, headingId) => {
+      const otherPages = pages.filter(otherPage => {
+        const headingMatch = (!headingId || otherPage._headers.find(hdg => hdg.id == headingId));
+        const slugMatch = otherPage._path.endsWith("/" + pathTail);
+        return slugMatch && headingMatch;
+      });
+      if (otherPages.length == 0) {
+        throw new Error(`No page exists with path tail '${pathTail}' and headingId '${headingId}' (from path '${page._path}')`);
+      } else if (otherPages.length > 1) {
+        //there are multiple matching pages -- try disambiguating by picking best match
+        otherPages.sort((a, b) => commonLength(b._path, page._path) - commonLength(a._path, page._path));
+        const firstChoice = otherPages[0];
+        const secondChoice = otherPages[1];
+        if (commonLength(firstChoice._path, page._path) > commonLength(secondChoice._path, page._path)) {
+          return firstChoice;
+        }
```

Comment:
We've sorted by best match, but we still need the first choice to be _better_ than the second.